modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageLeaderElectionListener.java

```diff
@@ -179,6 +197,12 @@ private interface Action {
      * Executes the given action if the current node is the Meta Storage leader.
      */
     private void execute(Action action) {
+        if (leaderSecondaryDutiesPaused.getAsBoolean()) {
```

Comment:
Why not inside the busyLock?

Reply:
We don't need protection provided by the busy lock to check whether secondary duties are paused; and without checking the lock, we can leave the method faster if they are paused. So I don't see why we would need to do it under the lock.

Reply:
We have a common template: 
Take BusyLock, if successful, then do any other checks and code. 
Let's try to stick to one template?

Reply:
No pattern is applicable in all situations. If it's not the best solution in some case, we should not stick to it just because it is a pattern and it works well in many other situations. In this situation, it is beneficial to do the check outside the lock (I explained why above), so I don't see why the pattern is to be blindly obeyed.